staging/src/k8s.io/apiserver/pkg/util/webhook/webhook.go

```diff
@@ -131,20 +131,24 @@ func WithExponentialBackoff(ctx context.Context, initialBackoff time.Duration, w
 		Steps:    5,
 	}
 
-	var err error
-	wait.ExponentialBackoff(backoff, func() (bool, error) {
-		err = webhookFn()
+	var webhookErr error
+	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
+		webhookErr = webhookFn()
 		if ctx.Err() != nil {
 			// we timed out or were cancelled, we should not retry
-			return true, err
+			return true, webhookErr
```

Comment:
if the context fails, the webhookErr could be nil and we would still want to fail, right?

Reply:
Yes, if context fails, we should return `ctx.Err()`